src/transfers/MultiAppMooseOkapiTransfer.C

```diff
@@ -169,50 +173,128 @@ MultiAppMooseOkapiTransfer::execute()
           MutableCoefficientsInterface & to_object =
               (this->*getSubAppObject)(_multi_app->appProblemBase(I), _multi_app_object_name, 0);
 
-          // Initialize an array to hold the expansion coefficients we'll receive
-          // from OpenMC. This array holds all of the expansion coefficients
-          // for the cell we specify. For a generic Zernike-Legendre expansion,
-          // the order used by OpenMC is specified in the tallies XML file. It is
-          // assumed that this XML order matches the available slots to write in
-          // the to_aux_scalar variables provided from MOOSE, so this is used to
-          // determine how large of an array to specify here that will be written.
-          // A check is provided in OpenMC to make sure that the length of the
-          // t % coeffs array) matches what is allocated in C++ to receive it.
+          int32_t * filter_indices = nullptr;
+          int32_t num_filter_indices;
+          int err_get =
+              openmc_tally_get_filters(_tally_index, &filter_indices, &num_filter_indices);
+          ErrorHandling::openmc_tally_get_filters(err_get, "MultiAppMooseOkapiTransfer");
+
+          if (num_filter_indices != 2)
+            mooseError(
+                "We expect there to be exactly two filters, one a cell filter and the other an "
+                "expansion filter. Check the tallies XML file to verify the existence of both for "
+                "the requested tally");
+          std::map<int32_t, int32_t> index_to_id;
+          for (decltype(n_cells) i = 0; i < n_cells; ++i)
+          {
+            int32_t id;
+            err_get = openmc_cell_get_id(i + 1, &id);
+            ErrorHandling::openmc_cell_get_id(err_get, "MultiAppMooseOkapiTransfer");
+            index_to_id[i + 1] = id;
+          }
+
+          // Important variables
+          int32_t order;
+          char type[20];
+          int32_t * cell_indices = nullptr;
+          int32_t num_cells_in_filter;
+          bool cell_id_found = false;
+          int32_t stride_integer = 0;
+          int32_t cell_filter_index;
+
+          std::string cell_filter_name = "cell";
```

Comment:
I remember trying to put checks like this in the constructor, but I ran into problems with `openmc_init()` not having yet been called. Do you think these checks should be done for every transfer? If not, is it possible to move them somewhere so that they're only called once? In the grand scheme of things the extra cost of redoing the checks every time is probably small since it'd be swamped by the MC cost, but I think it might be cleaner to emphasize that this is a set-up type of check.

Reply:
Thanks for the reminder nudge, I think the newest commit looks great!